### ISSUE TITLE
System crash using Wekan

### ISSUE THREAD

**Issue by Kisakila** [wekan/wekan#2777]
Hello,
Wekan has been installed on my Debian 10 server using SNAP. 
My server is offline, i mean it has no internet access. Wekan has been set up using "snap ack" and "snap install"

I can use it (the wekan app) correctly for a couple of hours, create cards, etc. "Randomly" the whole system crashes. 

Not even a response to PING.

The only solution is a "hard reboot". Otherwise, nothing else is possible.

I tried to find something in syslog but nothing weird.

Have you ever seen that ? any idea about a logfile that could help me to figure that out ?

i'm running out of ideas... 

thanks you all for your help.

**Comment by xet7** [comment]
What does these show? Please anonymize logs before adding.
```
sudo snap logs wekan.wekan

sudo snap logs wekan.mongodb
```

**Comment by Kisakila** [comment]
Name   Version          Rev   Tracking  Publisher   Notes
core   16-2.42          7917  -         canonical✓  core
wekan  3.49-1-g13a2bd6  647   -         xet7        -

---------------------------------------------------
RAM : 2Go
---------------------------------------------------
wekan.mongodb - A LOT of these : 

wekan3server mongod.27019[30636]: [conn5] received client metadata from 127.0.0.1:40722 conn5: { driver: { name: "nodejs", version: "2.2.9" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "4.19.0-6-amd64" }, platform: "Node.js v8.16.1, LE, mongodb-core: 2.0.11" }

---------------------------------------------------

cleaning the wekan.wekan and i'll send it here.
thanks a lot !

**Comment by xet7** [comment]
@Kisakila 

When you login to your server, is there any swap used? You see it with:
```
free -h
```

**Comment by Kisakila** [comment]
The problem is the complete crash.
all i can do is a hard reset and of course after that, everything is fine.

i dont even have time to check on the RAM and swap on the system.

i'll add more ram, just in case (let's say 4Go in total). to be honnest sometimes it can be a bit "slow". i didn't expect the whole to crash so quickly.

i'll do it on monday and let you know

**Comment by xet7** [comment]
@Kisakila 

How many users are using Wekan at once?

Do you have SSD or NVME disk at server?

**Comment by xet7** [comment]
Is there any errors at browser right click / Inspect / Console ?

**Comment by xet7** [comment]
I'm just thinking how it could freeze whole server that it does not respond to ping.

I run latest Debian at my laptop, with this sources.list:
```
# 

# deb cdrom:[Debian GNU/Linux 10.0.0 _unstable_ - Official amd64 NETINST 20190706-10:23]/ unstable contrib main non-free

# deb cdrom:[Debian GNU/Linux 10.0.0 _unstable_ - Official amd64 NETINST 20190706-10:23]/ unstable contrib main non-free

# original, commented out when installing cegui
deb [arch=amd64,i386] http://www.nic.funet.fi/debian/ testing main non-free contrib
deb-src [arch=amd64,i386] http://www.nic.funet.fi/debian/ testing main non-free contrib

# cegui
# deb http://www.nic.funet.fi/debian/ unstable main non-free contrib
# deb-src http://www.nic.funet.fi/debian/ unstable main non-free contrib

deb [arch=amd64,i386] http://security.debian.org/debian-security buster/updates main contrib non-free
deb-src [arch=amd64,i386] http://security.debian.org/debian-security buster/updates main contrib non-free

# unstable-updates, previously known as 'volatile'
deb [arch=amd64,i386] http://www.nic.funet.fi/debian/ buster-updates main contrib non-free
deb-src [arch=amd64,i386] http://www.nic.funet.fi/debian/ buster-updates main contrib non-free

# backports
# deb http://www.nic.funet.fi/debian/ unstable-backports main contrib non-free

# This system was installed using small removable media
# (e.g. netinst, live or single CD). The matching "deb cdrom"
# entries were disabled at the end of the installation process.
# For information about how to configure apt package sources,
# see the sources.list(5) manual.
```
Currently I'm using i5 HP laptop with SSD and 8  GB RAM. It has Wekan Snap, XFCE desktop, browsers, etc. I did also use same SSD in my Core 2 Duo laptop that has 8 GB RAM. I have also had Wekan running at RasPi3b+ that has Ubuntu 18.04 arm64 server and 1 GB RAM without X desktop, and that also was usable for my single user use.

But I have not seen such system crash that server would not respond to ping. It just makes me first think about hardware problem.

Is there any errors with dmesg ? Or is the SSD disk OK, is there smart monitoring errors.

**Comment by Kisakila** [comment]
hello.
i'm  not sure about the hardware problem.
wekan is running on a brand new virtual machine (esxi).
Nothing weird with the disk.
This VM has been rebuilt twice, just in case. 
When i try on my virtualbox, i don't have the same issue. 

i'm currently trying again, im monitoring dmesg, ram, and process and i'll let u know if anything happens.

**Comment by Kisakila** [comment]
Message from syslogd@severwekan at Nov  4 ...
 kernel:[ 3364.252236] Kernel panic - not syncing: Fatal exception in interrupt
and 
DMESG

DMESG
[Nov 4 00:01] ------------[ cut here ]------------
[  +0.000004] kernel BUG at drivers/net/vmxnet3/vmxnet3_drv.c:1441!
[  +0.000074] invalid opcode: 0000 [#1] SMP PTI
[  +0.000033] CPU: 1 PID: 0 Comm: swapper/1 Not tainted 4.19.0-6-amd64 #1 Debian 4.19.67-2+deb10u1
[  +0.000050] Hardware name: VMware, Inc. VMware Virtual Platform/440BX Desktop Reference Platform, BIOS 6.00 04/05/2016
[  +0.000086] RIP: 0010:vmxnet3_rq_rx_complete+0x9e0/0xe70 [vmxnet3]

**Comment by Kisakila** [comment]
https://kb.vmware.com/s/article/2151480

it seems to be the answer !
i don't close the case yet.... still testing ;) 

if it's ok i'll let you know and close

**Comment by xet7** [comment]
Wekan does not work where kernel is limited, like OpenVZ, OVH, Kimsufi:
https://github.com/wekan/wekan/wiki/Platforms#other-clouds-can-have-some-restrictions-like-requires-from-source-install-has-restricted-2x-kernel-does-not-support-websockets-or-not-tested-yet

In that case Snap and Docker have problem, so you need to install from bundle:
https://github.com/wekan/wekan/wiki/Platforms#not-exposed-to-internet-bundle-for-raspi-3-arm64-windows-and-any-nodemongo-cpu-architectures-no-automatic-updates-no-sandboxing

Wekan works on KVM, Xen, HyperV.

**Comment by Kisakila** [comment]
Time for me to close the case. 
It's working with this method : https://kb.vmware.com/s/article/2151480
Thanks again for your help and support guys. 

Cheers !

**Comment by tezzcan** [comment]
> Time for me to close the case. It's working with this method : https://kb.vmware.com/s/article/2151480 Thanks again for your help and support guys.
> 
> Cheers !

Our problem was very similar: 

Log says:
`Feb 17 17:35:31 download3 kernel: [  210.518544] ------------[ cut here ]------------
Feb 17 17:35:31 download3 kernel: [  210.518547] kernel BUG at drivers/net/vmxnet3/vmxnet3_drv.c:1441!
Feb 17 17:35:31 download3 kernel: [  210.518613] invalid opcode: 0000 [#1] SMP PTI
Feb 17 17:35:31 download3 kernel: [  210.518633] CPU: 3 PID: 2330 Comm: python Not tainted 5.4.0-99-generic #112-Ubuntu`

We tried different kernel versions 5.4, 5.10 and 5.16. That changed nothing. We have fixed the problem because of your response by link.

So thank you.
